src/main/java/dbaccessor/userchat/UserChatAccessor.java

```diff
@@ -50,4 +51,49 @@ public boolean checkIfUserChatIdExists(long userId, long chatId) {
     
         return !resultSet.isEmpty();
     }
+
+    /**
+     * Finds the UserId corresponding to the given username.
+     * If a Chat exists between the two Users, returns its ChatId and the UserId found above.
+     * Otherwise, returns a zero ChatId along with the UserId found above. 
+     */
+    public CreateChat.ChatIdWithUserIds getChatIdWithUserIdsIfExistsBetweenUsers(long userId, String username) {
+        String sqlStatment = "SELECT " 
+                                + "ChatsOfUserID2.ChatID AS ChatID, " 
+                                + "UserChat.UserID AS UserID1, "
+                                + "ChatsOfUserID2.UserID2 AS UserID2, "
+                            + "FROM (" 
+                                + "SELECT " 
+                                    + "SecondUserID.UserID2 AS UserID2, " 
+                                    + "UserChat.ChatID AS ChatID " 
+                                + "FROM (" 
+                                    + "SELECT UserID AS UserID2 " 
+                                    + "FROM User " 
+                                    + "WHERE Username = @username" 
+                                    + ") AS SecondUserID " 
+                                + "LEFT OUTER JOIN UserChat " 
+                                + "ON SecondUserID.UserID2 = UserChat.UserID" 
+                                + ") AS ChatsOfUserID2 " 
+                            + "LEFT OUTER JOIN UserChat " 
+                            + "ON ChatsOfUserID2.ChatID = UserChat.ChatID";
+                            // + "WHERE UserChat.UserID = @userId";
+
+        Statement statement = Statement.newBuilder(sqlStatment).bind("username").to(username).bind("userId").to(userId).build();
+        List<CreateChat.ChatIdWithUserIds> resultSet = spannerTemplate.query(
+                                                        CreateChat.ChatIdWithUserIds.class, 
+                                                        statement, 
+                                                        new SpannerQueryOptions().setAllowPartialRead(true)
+                                                        );
+
+        for (CreateChat.ChatIdWithUserIds result : resultSet) {
+            if (result.getUserId1() == userId) {
+                return result;
+            }
+        }
+
+        resultSet.get(0).setChatId(0);
```

Comment:
why are we setting these to 0?

Reply:
If we get a `result` in the above `for` loop in which `result.getUserId1() == userId`, it means that there exists a `Chat` between the two `Users`. If it doesn't exist, we might still get back `result`(s) in which the `ChatID` may not be `null` i.e. `0`. These entries correspond to `Chats` of `User`2 with `Users` other than `User`1. Since we want to return a non-zero `ChatID` of only the `Chat` between `User`1 and `User`2 and a zero `ChatID` if this `Chat` doesn't exist, hence we set the `ChatID` to zero before returning the `result`.